### PR TITLE
upgrade runtime controller to 0.15

### PR DESCRIPTION
**Pull request by ruanxin**

### CODE REVIEW

File: pkg/util/error.go [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+	"errors"
+	"strings"
+
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/discovery"
+)
+
+func IsNotFound(err error) bool {
```
Comment (ruanxin) [inline pkg/util/error.go:13]: To reviewer:
I unified `k8serrors.IsNotFound` with this customized function, based on the workaround solution mentioned in this open issue  https://github.com/kubernetes-sigs/controller-runtime/issues/2354 thanks @LeelaChacha bring this into attention

**Comment by lindnerby** [comment]
@ruanxin Can you rebase main and delete the temporary unregistering of metrics with this PR as well please?

**Review by lindnerby — DISMISSED** [review]
lgtm

**Review by lindnerby — APPROVED** [review]

**Pull request merged**